### DISCUSSION TITLE
Is it possible to set baseURL?

### DISCUSSION THREAD

**Discussion by Kazuhiro-Mimaki** [mswjs/msw#1696] [Q&A] [answered]
Is it possible to set baseURL?

For example,

```ts
const handlers = [
  rest.get('http://example.com/xxx', () => {}),
  rest.get('http://example.com/yyy', () => {}),
  rest.get('http://example.com/zzz', () => {}),
]
```
It is good idea to define module "baseURL".

```ts
const baseURL = (path: `/${string}`) => `http://example.com${path}`

const handlers = [
  rest.get(baseURL('/xxx'), () => {}),
  rest.get(baseURL('/yyy'), () => {}),
  rest.get(baseURL('/zzz'), () => {}),
]
```

However, I think it is simpler to write the interface like following.

```ts
rest.baseURL = 'http://example.com'

const handlers = [
  rest.get('/xxx', () => {}),
  rest.get('/yyy', () => {}),
  rest.get('/zzz', () => {}),
]
```

Is it possible?

**Comment by kettanaito** [comment, marked as answer]
Hi, @Kazuhiro-Mimaki. Please take a look at the existing discussions around this:

- https://github.com/mswjs/msw/issues/397
- https://github.com/mswjs/msw/discussions/1636

TL;DR No, MSW provides no means to set a base URL using its API. You should set the base URL by yourself, and the issues I linked above illustrate a few ways how to do that.
